lib/exprengine.cpp

```diff
@@ -2646,19 +2647,30 @@ static std::string execute(const Token *start, const Token *end, Data &data)
                         data.assignStructMember(tok2, &*structVal, memberName, memberValue);
                         continue;
                     }
+
+                    // Assign a pointer value
                     if (lhs->isUnaryOp("*") && lhs->astOperand1()->varId()) {
                         const Token *varToken = tok2->astOperand1()->astOperand1();
                         ExprEngine::ValuePtr val = data.getValue(varToken->varId(), varToken->valueType(), varToken);
-                        if (val && val->type == ExprEngine::ValueType::ArrayValue) {
-                            // Try to assign "any" value
-                            auto arrayValue = std::dynamic_pointer_cast<ExprEngine::ArrayValue>(val);
-                            arrayValue->assign(std::make_shared<ExprEngine::IntRange>("0", 0, 0), std::make_shared<ExprEngine::BailoutValue>());
+                        if (!val || val->type != ExprEngine::ValueType::ArrayValue)
```

Comment:
At some point we want to handle `AddressOf` also..

    void foo() {
        int a = 2;
        int*p = &a;
        while (1) *p = 0;
        return a == 2;  // <- should generate a z3::unsat
    }

Reply:
Good point. I implemented this.